app/views/products/_form.html.haml

```diff
@@ -8,9 +8,9 @@
 			= f.number_field :price, :class => 'text_field', :step => 'any'
 			%br/
 			= f.label :quantity
-			= f.number_field :quantity, :value => 1, :min => 0
+			= f.number_field :quantity, :value => @product.quantity || 1, :min => 0
```

Comment:
Why do you need to set the value here manually? Could you not just let the `min` be 1?